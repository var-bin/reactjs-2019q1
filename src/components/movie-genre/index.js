export { MovieGenre } from './movie-genre';
