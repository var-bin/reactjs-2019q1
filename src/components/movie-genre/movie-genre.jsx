import React from 'react';

export const MovieGenre = (props) => {
  const {
    genre
  } = props;

  return (
    <a href={`#${genre}`} className="is-inline-block">
      &nbsp;#
      {genre}
    </a>
  );
};
