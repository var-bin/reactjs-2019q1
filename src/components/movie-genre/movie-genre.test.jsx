import React from 'react';
import { shallow } from 'enzyme';

import { MovieGenre } from './movie-genre';

describe('MovieGenre component:', () => {
  let wrapper;
  const genre = 'my awesome genre';

  beforeEach(() => {
    wrapper = shallow(
      <div>
        <MovieGenre genre={genre} />
      </div>
    );
  });

  it('Should render without errors:', () => {
    expect(wrapper.find(MovieGenre)).toHaveLength(1);
  });

  it('Should have right props:', () => {
    const { genre: localGenre } = wrapper.find(MovieGenre).props();

    expect(localGenre).toEqual(genre);
  });
});
