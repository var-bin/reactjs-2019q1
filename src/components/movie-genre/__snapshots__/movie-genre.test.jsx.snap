// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGenre component: Should have right text: 1`] = `"<a href=\\"#my awesome genre\\" class=\\"is-inline-block\\">Â #my awesome genre</a>"`;
